lib/liquid.rb

```diff
@@ -24,6 +24,8 @@ module Liquid
   ArgumentSeparator           = ','
   FilterArgumentSeparator     = ':'
   VariableAttributeSeparator  = '.'
+  WhiteSpaceControl           = /-/
+  WhiteSpace                  = /[ \t]*\r?\n?[ \t]*/
```

Comment:
I'm not against a full trim. If anything the more it cleans up the whitespace the better. How useful that is i'm not sure. Maybe if your designing a template and you are using empty lines to break apart different blocks